### PR TITLE
P5-17: Analysis of distance to target

### PR DESCRIPTION
**Pull request by C0DK**
Mangler korrekturlæsning

### CODE REVIEW

File: report/2.Analysis/DistanceCalculation.tex [hunk 1]
```diff
@@ -0,0 +1,34 @@
...
+Based on the visual-feed, it should be possible to calculate the movement of the target, but this leaves the problem of finding the distance between target and device.
+
+
+A couple of different solutions could be imagined.
+One solution, would be using a 3D camera, while another would be to use a Ultrasonic Sensor, as presented in section \ref{sec:sensors}, and point it at the target.
+Using an ultrasonic sensor is the cheapest option and probably has a higher degree of precision, but this would require the device to actually point the sensor at the target, which is quite time costly.
+% !! Hvordan får jeg kilde på ovenstående sætning? !!
+However using this to reliably predict where to shoot, can still be difficult, depending on projectile.
+An actual gun, would shoot bullets quite fast and reliably, however using this seems both dangerous and impossible.
+Using a LEGO gun or cannon would more feasible, but these are less consistent in trajectory and speed, sot his would be the most obvious choice.
```
Comment (nillas12) [inline report/2.Analysis/DistanceCalculation.tex]: these have less consistent trajectory and speed, so this (both change are (i think) remove in, and fix the spacing on sot his.

File: report/2.Analysis/DistanceCalculation.tex [hunk 2]
```diff
@@ -0,0 +1,34 @@
...
+LEGO projectiles, and other toy weapons, will have less speed, as it is created for children, and the higher speed the more dangerous.
+This can present itself as quite a big hurdle, in a project that already seems quite hard from a software perspective, without including the hardware complications of a gun and the lack of predictability.
```
Comment (nillas12) [inline report/2.Analysis/DistanceCalculation.tex]: I don't like the first part, but also don't know what to change it to. It is probably fine.

File: report/2.Analysis/DistanceCalculation.tex [hunk 3]
```diff
@@ -0,0 +1,34 @@
...
+
+
+A couple of different solutions could be imagined.
+One solution, would be using a 3D camera, while another would be to use a Ultrasonic Sensor, as presented in section \ref{sec:sensors}, and point it at the target.
+Using an ultrasonic sensor is the cheapest option and probably has a higher degree of precision, but this would require the device to actually point the sensor at the target, which is quite time costly.
+% !! Hvordan får jeg kilde på ovenstående sætning? !!
+However using this to reliably predict where to shoot, can still be difficult, depending on projectile.
+An actual gun, would shoot bullets quite fast and reliably, however using this seems both dangerous and impossible.
+Using a LEGO gun or cannon would more feasible, but these are less consistent in trajectory and speed, sot his would be the most obvious choice.
+LEGO projectiles, and other toy weapons, will have less speed, as it is created for children, and the higher speed the more dangerous.
```
Comment (nillas12) [inline report/2.Analysis/DistanceCalculation.tex]: Again, I don't really like the "more dangerous" argument. This time I don't know what to change it to tho.

File: report/2.Analysis/DistanceCalculation.tex (hunk 2 above)
Comment (nillas12) [inline report/2.Analysis/DistanceCalculation.tex]: remote quite from "seems quite hard" (thats what she said)

File: report/2.Analysis/DistanceCalculation.tex [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+This can present itself as quite a big hurdle, in a project that already seems quite hard from a software perspective, without including the hardware complications of a gun and the lack of predictability.
+
+
+A great alternative, branching out of the Utrasonic Sensor solution, would be to simply use a visible laser and aim at the target.
```
Comment (nillas12) [inline report/2.Analysis/DistanceCalculation.tex]: I was about to write something about visible laser, but then I realise infrared.
Insted: to simply point a visible laster at the target.

File: report/2.Analysis/DistanceCalculation.tex [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+The issue of distance, is based on the fact that a given projectile takes a certain amount of time to reach the target, and this introduces uncertainty. 
+However, using a laser will reduce travel-time greatly, as the projectile, being the light beam, will move at the speed of light.
```
Comment (nillas12) [inline report/2.Analysis/DistanceCalculation.tex]: I think reduce travel-time greatly is the understatement of the year. Maybe rewrite this.

File: report/2.Analysis/DistanceCalculation.tex [hunk 6]
```diff
@@ -0,0 +1,34 @@
...
+As the tests of this project will be done in our group rooms, the distance between a target and the device will never be more than 10 meters, which means the travel speed of the projectile will be near-instant.
```
Comment (nillas12) [inline report/2.Analysis/DistanceCalculation.tex]: I think when we're talking 299792458 m/s we can say instant. (near is fine tho

File: report/2.Analysis/DistanceCalculation.tex [hunk 7]
```diff
@@ -0,0 +1,34 @@
...
+LEGO projectiles, and other toy weapons, will have less speed, as it is created for children, and the higher speed the more dangerous.
+This can present itself as quite a big hurdle, in a project that already seems quite hard from a software perspective, without including the hardware complications of a gun and the lack of predictability.
+
+
+A great alternative, branching out of the Utrasonic Sensor solution, would be to simply use a visible laser and aim at the target.
+The issue of distance, is based on the fact that a given projectile takes a certain amount of time to reach the target, and this introduces uncertainty. 
+However, using a laser will reduce travel-time greatly, as the projectile, being the light beam, will move at the speed of light.
+As the tests of this project will be done in our group rooms, the distance between a target and the device will never be more than 10 meters, which means the travel speed of the projectile will be near-instant. 
+This means the intersection point is going to be so close to the target, that the distance will be irrelevant.
+Because of the above reasoning, it makes sense to use a laser initially.
```
Comment (nillas12) [inline report/2.Analysis/DistanceCalculation.tex]: Maybe instead use: Based on the above reasoning?

File: report/2.Analysis/DistanceCalculation.tex [hunk 8]
```diff
@@ -0,0 +1,34 @@
+% !TeX root = ../main.tex
+% konkluder at der er lettere at bruge en laser. 
+% hen af vejen overvej de udregner der kan bruges til at udregne afstanden, i tilfælde af at vi får ekstra tid
+
+\section{Distance to target}
+Being able to successfully hit a moving target, relies on the speed and acceleration of the target, as well as the speed and acceleration of the projectile.
```
[the remark this answers is not in the document]
Comment (C0DK) [reply report/2.Analysis/DistanceCalculation.tex]: well it's an opener for what we will discuss in this section. seems relevant. i can restructure the sentence tho

File: report/2.Analysis/DistanceCalculation.tex [hunk 9]
```diff
@@ -0,0 +1,34 @@
...
+Shooting a projectile at the target, depending on the speed of the projectile, it might take multiple milliseconds or even seconds for the projectile to hit the target.
+But as the target is moving, it will no longer be the location at which the projectile was shot towards.
+Because of this, the projectile has to aim at where the target is moving, not where it is at.
+This causes some problems regarding movement time of the motors on the device, which was covered in section \ref{sec:rts-intro}, but also problems regarding the intersection of target and projectile.
+This problem is a matter of finding the distance in front of the target to a given intersection point, based on the speed and acceleration of the target, that matches the time it takes for the projectile to reach a given intersection point.
```
[the remark this answers is not in the document]
Comment (C0DK) [reply report/2.Analysis/DistanceCalculation.tex]: sætningen er ændret en del. sig om det er bedre.

File: report/2.Analysis/DistanceCalculation.tex [hunk 10]
```diff
@@ -0,0 +1,34 @@
+% !TeX root = ../main.tex
+% konkluder at der er lettere at bruge en laser. 
+% hen af vejen overvej de udregner der kan bruges til at udregne afstanden, i tilfælde af at vi får ekstra tid
+
+\section{Distance to target}
+Being able to successfully hit a moving target, relies on the speed and acceleration of the target, as well as the speed and acceleration of the projectile.
+Shooting a projectile at the target, depending on the speed of the projectile, it might take multiple milliseconds or even seconds for the projectile to hit the target.
```
[the remark this answers is not in the document]
Comment (C0DK) [reply report/2.Analysis/DistanceCalculation.tex]: several er skrevet istedet.

File: report/2.Analysis/DistanceCalculation.tex (hunk 8 above)
[the remark this answers is not in the document]
Comment (Moesgaarda) [reply report/2.Analysis/DistanceCalculation.tex]: I think it's pretty good for an opening.

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 5e883f22eb36e992988587bc2a362064b82e898b
  * Invalid type. Valid types are **feat, fix, docs, style, refactor, perf, test, chore, revert, fixes, adds, Fixes, Adds, Reverts, Feat, Fix, Docs, Style, Refactor, Refactors**

Guidelines are available at https://github.com/aau-hcl-p5/main

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 5e883f22eb36e992988587bc2a362064b82e898b
  * Invalid type. Valid types are **feat, fix, docs, style, refactor, perf, test, chore, revert, fixes, adds, Fixes, Adds, Reverts, Feat, Fix, Docs, Style, Refactor, Refactors**

 * Commit: 3f65626be11575dde53c51df1bf7bb7ff27a202e
  * Commits must be in the following format: **%{type}: %{description}**

Guidelines are available at https://github.com/aau-hcl-p5/main

---
This message was auto-generated by https://gitcop.com

**Comment by GitCop** [comment]
There were the following issues with your Pull Request

 * Commit: 5e883f22eb36e992988587bc2a362064b82e898b
  * Invalid type. Valid types are **feat, fix, docs, style, refactor, perf, test, chore, revert, fixes, adds, Fixes, Adds, Reverts, Feat, Fix, Docs, Style, Refactor, Refactors**

 * Commit: 3f65626be11575dde53c51df1bf7bb7ff27a202e
  * Commits must be in the following format: **%{type}: %{description}**

Guidelines are available at https://github.com/aau-hcl-p5/main

---
This message was auto-generated by https://gitcop.com

**Pull request merged**